core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -119,7 +120,9 @@ private[chisel3] trait HasId extends InstanceId {
     * @param seed Seed for the name of this component
     * @return this object
     */
-  private [chisel3] def autoSeed(seed: String): this.type = {
+  private[chisel3] def autoSeed(seed: String): this.type = forceAutoSeed(seed)
+  // Bypass the overridden behavior of autoSeed in [[Data]], apply autoSeed even to ports
+  private[chisel3] def forceAutoSeed(seed: String): this.type = {
```

Comment:
Is the plan to remove autoseed in the future?

Reply:
Possibly, a good question for the dev meeting